source/GameAction.cpp

```diff
@@ -0,0 +1,228 @@
+/* GameAction.cpp
+Copyright (c) 2020 by Jonathan Steck
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "GameAction.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Dialog.h"
+#include "Format.h"
+#include "GameData.h"
+#include "GameEvent.h"
+#include "PlayerInfo.h"
+#include "Random.h"
+
+#include <cstdlib>
+#include <vector>
+
+using namespace std;
+
+
+
+// Construct and Load() at the same time.
+GameAction::GameAction(const DataNode &node, const string &missionName)
+{
+	Load(node, missionName);
+}
+
+
+
+void GameAction::Load(const DataNode &node, const string &missionName)
+{
+	empty = false;
+	for(const DataNode &child : node)
+		LoadSingle(child, missionName);
+}
+
+
+
+// Load a single child at a time, used for streamlining MissionAction::Load.
+void GameAction::LoadSingle(const DataNode &child, const string &missionName)
+{
+	const string &key = child.Token(0);
+	bool hasValue = (child.Size() >= 2);
+	
+	if(key == "log")
```

Comment:
Needs to also check hasValue (since Token(1) is accessed).

Reply:
This is handled by the IsSpecial bool.

Reply:
You're right. I didn't read the code properly.